src/main/java/org/jenkinsci/remoting/util/AnonymousClassWarnings.java

```diff
@@ -39,19 +37,21 @@
 import java.util.Map;
 import java.util.WeakHashMap;
 import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 import java.util.concurrent.RejectedExecutionException;
 import java.util.logging.Logger;
 
 /**
  * Issues warnings about attempts to (de-)serialize anonymous, local, or synthetic classes.
+ *
  * @see <a href="https://jenkins.io/redirect/serialization-of-anonymous-classes/">More information</a>
  */
 public class AnonymousClassWarnings {
 
     private static final Logger LOGGER = Logger.getLogger(AnonymousClassWarnings.class.getName());
     private static final Map<Class<?>, Boolean> checked = new WeakHashMap<>();
 
-    private final static ExecutorService THREAD_POOL =  new AtmostOneThreadExecutor(new NamingThreadFactory(new DaemonThreadFactory(), AnonymousClassWarnings.class.getSimpleName()));
+    private static final ExecutorService THREAD_POOL = Executors.newFixedThreadPool(1, new NamingThreadFactory(Executors.defaultThreadFactory(), AnonymousClassWarnings.class.getSimpleName()));
```

Comment:
There are actually 2 changes there :
* 1 constant thread instead of 1 on-demand.
* Normal thread vs. daemon thread.

Could you check performance with only one or the other?

Reply:
Or could you check #559 to see if it works? If so, it is simpler.

Reply:
Tested with DaemonThreadFactory, result are also looks good. Response time 50-300ms.
PR updated